lib/util/webvpn_proxy.dart

```
@@ -118,16 +118,18 @@ class WebvpnProxy {
 
       debugPrint("Logging into WebVPN");
 
-      // Temporary cookie jar
-      IndependentCookieJar workJar = IndependentCookieJar();
-      loginSession =
-          UISLoginTool.loginUIS(dio, WEBVPN_LOGIN_URL, workJar, _personInfo);
-      await loginSession;
-
-      // Clone from temp jar to our dedicated webvpn jar
-      webvpnCookieJar.cloneFrom(workJar);
-      loginSession = null;
-      isLoggedIn = true;
+      try {
+        // Temporary cookie jar
+        IndependentCookieJar workJar = IndependentCookieJar();
+        loginSession =
+            UISLoginTool.loginUIS(dio, WEBVPN_LOGIN_URL, workJar, _personInfo);
+        await loginSession;
+        // Clone from temp jar to our dedicated webvpn jar
+        webvpnCookieJar.cloneFrom(workJar);
+        isLoggedIn = true;
+      } finally {
+        loginSession = null;
+      }
```

It will simply clean up and propagate any thrown exception. It this behavior expected?

---

Oh, I cut the `catch` and forgot to paste.

---

> Oh, I cut the `catch` and forgot to paste.

In fact, I didn't think you made a mistake at the beginning; I just wanted to confirm if this is what you intended. Actually, NOT ignoring the error feels more user-friendly...